crates/wasmtime/src/runtime/func.rs

```diff
@@ -1646,7 +1646,19 @@ fn enter_wasm<T>(store: &mut StoreContextMut<'_, T>) -> Option<usize> {
         return None;
     }
 
+    // When Cranelift has support for the host then we might be running native
+    // compiled code meaning we need to read the actual stack pointer. If
+    // Cranelift can't be used though then we're guaranteed to be running pulley
+    // in which case this stack poitner isn't actually used as Pulley has custom
+    // mechanisms for stack overflow.
+    #[cfg(has_cranelift_host_backend)]
```

Comment:
Can we rename this `has_host_compiler_backend`? While we don't have any target supported on Winch but not Cranelift now, it doesn't seem like it would be impossible in the future, so we should just try to be a little more precise about what we are saying.